SCClassLibrary/Common/UnitTesting/UnitTest.sc

```diff
@@ -33,24 +37,41 @@ UnitTest {
 		if(method.isNil) {
 			Error("Test method not found " + methodName).throw
 		};
-		class.new.runTestMethod(method);
+		this.use {
+			class.new.runTestMethod(method)
+		}
 	}
 
+	// this should be called when a new unit test instance is created
+	*setUpClass {}
+
+	// this should be called when a unit test instance is abandoned
+	*tearDownClass {}
 
 	// called before each test
 	setUp {}
 
 	// called after each test
 	tearDown {}
 
+	// call a function in the context of this test class
+	*use { |func|
```

Comment:
this should be private, and i would recommend a more descriptive name like `prRunWithinSetUpClass`

Reply:
hm, I have to think about this. `use` is a well established term for this kind of functionality. Also it doesn't have to be private – you could call it from anywhere and it would run correctly, wouldn't it?

Reply:
> use is a well established term for this kind of functionality.

even making an analogy to things like File and Environment, here there are 2 levels of "context" -- `setUpClass` and `setUp`. that leaves too much room for confusion.

> Also it doesn't have to be private – you could call it from anywhere and it would run correctly, wouldn't it?

well, not exactly anywhere, if `setUpClass` is not reentrant or if it requires `tearDownClass` to run before it is run again. it's not difficult to imagine this being the case if it is used to allocate complex resources. i can't really see a use for this outside the UnitTest class itself, allowing it to be run from user code could introduce subtle bugs, and it is far easier to make a private method public than vice versa, so the burden is on you to show why it should be public here.

Reply:
I see, the setUp on the object level is presupposed.